specs/userRegistration.spec.ts

```diff
@@ -0,0 +1,409 @@
+    const mockPromise = new Promise((resolve, reject) => {
+      setTimeout(() => resolve('User add success'), 20);
+    })
+    res.locals.hashedPwd = mockPwd;
+    const req = requestMock({}, { username: mockUser });
+
+    const spy = jest.spyOn(User, 'create').mockImplementation(() => mockPromise as any);
+
+    await postNewUser(req, res, next);
+    expect(spy).toHaveBeenCalledWith({
+      username: mockUser,
+      password: res.locals.hashedPwd
+    });
+    done();
+  });
+
+  test('it should try adding a new user', async (done) => {
+    const mockPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        const obj = {
+          _id: '1234',
+          username: 'guest',
+          password: mockPwd
+        };
+        resolve(obj);
+      }, 20);
+    })
+    const mockPwd = 'wasd';
+    const req = requestMock({}, { username: 'guest', password: mockPwd, password2: mockPwd, potato: 'potato' });
+
+    const addUser = jest.spyOn(User, 'create').mockImplementation(() => mockPromise as any);
+
+    await postNewUser(req, res, next);
+
+    expect(addUser).toHaveBeenCalled();
+    done();
+  });
+
+  test('it should return a response with JSON if it succeeds', async (done) => {
+    const mockUser = 'guest';
+    const mockPwd = 'aaaaaaaaaaaaaa';
+    const mockPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        const obj = {
+          _id: '1234',
+          username: mockUser,
+          password: mockPwd
+        }
+        resolve(obj);
+      }, 20);
+    })
+    res.locals.hashedPwd = mockPwd;
+    const req = requestMock({}, { username: mockUser });
+
+    jest.spyOn(User, 'create').mockImplementation(() => mockPromise as any);
+
+    await postNewUser(req, res, next);
+
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.json).toHaveBeenCalledWith(
+      expect.objectContaining({
+        msg: expect.any(String),
+        user: expect.objectContaining({
+          _id: expect.anything(),
+          username: expect.any(String)
+        })
+      })
+    );
+    done();
+  });
+
+  test('it should return a response with JSON if it fails', async (done) => {
+    const mockUser = 'guest';
+    const mockPwd = 'aaaaaaaaaaaaaa';
+    const mockPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        reject(new Error('mock error msg'));
+      }, 20);
+    });
+    res.locals.hashedPwd = mockPwd;
+    const req = requestMock({}, { username: mockUser });
+
+    jest.spyOn(User, 'create').mockImplementation(() => mockPromise as any);
+    jest.spyOn(console, 'error').mockImplementation(() => null);
+
+    await postNewUser(req, res, next);
+
+    expect(res.status).toHaveBeenCalledWith(500);
+    expect(res.json).toHaveBeenCalledWith(
+      expect.objectContaining({
+        msg: expect.any(String),
+      })
+    );
+    done();
+  });
+
+  test('it should send a response starting with "Error":', async (done) => {
```

Comment:
A small comment, but I think this test description could be a little more descriptive to say what is expected to trigger the error response.

Reply:
Changed to: `'it should send a response starting with "Error:" if it fails'`